test/lib/jdk/test/lib/util/CoreUtils.java

```diff
@@ -152,6 +152,30 @@ public static String getCoreFileLocation(String crashOutputString) throws IOExce
                     line = line.trim();
                     System.out.println(line);
                     if (line.startsWith("|")) {
+                        if (line.split("\s", 2)[0].endsWith("systemd-coredump")) {
+                            // A systemd linux system. Try to retrieve core
+                            // file. It can take a few seconds for the system to
+                            // process the just produced core file so we may need to
+                            // retry a few times.
+                            System.out.println("Running systemd-coredump: trying coredumpctl command");
+                            String core = "core";
+                            try {
+                                for (int i = 0; i < 10; i++) {
+                                    Thread.sleep(5000);
+                                    OutputAnalyzer out = ProcessTools.executeProcess("coredumpctl", "dump",  "-1",  "-o", core, Long.valueOf(pid).toString());
+                                    if (!out.getOutput().contains("output may be incomplete")) {
+                                        break;
+                                    }
+                                }
+                            } catch(Throwable t) {
+                            }
+                            final File coreFile = new File(core);
+                            if (coreFile.exists()) {
+                                Asserts.assertGT(coreFile.length(), 0L, "Unexpected core size");
+                                System.out.println("coredumpctl succeeded");
+                                return core;
```

Comment:
At the start of the outermost if/else block, there is a comment that says "See if we can figure out the likely reason the core file was not found". However, now we are doing further attempts to find the core file. At the very least the comment should be updated. Maybe add something like "Recover from failure if possible."

Reply:
ok.